features/collection/persistence-rrd/src/main/java/org/opennms/netmgt/collection/persistence/rrd/RrdPersistOperationBuilder.java

```diff
@@ -211,7 +222,9 @@ public void commit() throws PersistException {
             List<RrdDataSource> dataSources = getDataSources();
             if (dataSources != null && dataSources.size() > 0) {
                 createRRD(m_rrdStrategy, ownerName, absolutePath, m_rrdName, getRepository().getStep(), dataSources, getRepository().getRraList(), m_metaData);
-                updateRRD(m_rrdStrategy, ownerName, absolutePath, m_rrdName, m_timeKeeper.getCurrentTime(), getValues());
+                long rrdTime = Boolean.getBoolean("org.opennms.netmgt.collection.persistence.rrd.useCollectionStartTime")
+                    ? m_collectionTime : m_timeKeeper.getCurrentTime();
```

Comment:
I would opt for overriding the TimeKeeper for the CollectionResource objects in the CollectionSet instead of explicitly passing the time through the persistence layers and performing this logic here.